docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,322 @@
+
+### oras pull 
+
+Pull a repository and display its metadata as formatted JSON in standard output. The following fields should be formatted in a JSON output:
+
+- `Ref`: full artifact reference by digest, e.g, `$REGISTRY/$REPO@$DIGEST`
+-  `Files`: a list of downloaded files
+    -  `Path`: the absolute file path of the pulled file (layer)
+    -  `Ref`: full reference by digest of the pulled file (layer)
+    -  `MediaType`: media type of the pulled file (layer) 
+    -  `Digest`: digest of the pulled file (layer) 
+    -  `Size`: file size in bytes
+
+Pull a repository that contains multiple layers (files) and show their descriptor metadata as pretty JSON in standard output.
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --artifact-type example/sbom sbom.spdx  --artifact-type example/vul-scan vul-scan.json --format json
+```
+
+```json
+{
+  "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186111",
+  "Files": [
+    {
+      "Path": "/home/user/oras-install/sbom.spdx",
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "MediaType": "application/vnd.oci.image.manifest.v1+json",
+      "Digest": "sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "Size": 820,
+      "Annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+    },
+    {
+      "Path": "/home/user/oras-install/vul-scan.json",
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b",
+      "MediaType": "application/vnd.oci.image.manifest.v1+json",
+      "Digest": "sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b",
+      "Size": 820,
+      "Annotations": {
+        "org.opencontainers.image.title": "vul-scan.json"
+      }
+    }
+  ]
+}
+```
+
+Pull a repository that contains multiple layers (files) and display its descriptor metadata as raw JSON in standard output.
+
+```bash
+oras pull $REGISTRY/$REPO:$TAG --format '{{toRawJson .}}'
+```
+
+```
+{"Ref":"localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186111","Files":[{"Path":"/home/user/oras-install/sbom.spdx","Ref":"localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222","MediaType":"application/vnd.oci.image.manifest.v1+json","Digest":"sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222","Size":820},{"Path":"/home/user/oras-install/vul-scan.json","Ref":"localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b","MediaType":"application/vnd.oci.image.manifest.v1+json","Digest":"sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b","Size":820}]}
+```
+
+### oras push
+
+Push two files to a repository and show the descriptor of the image manifest in pretty JSON format.
```

Comment:
Same question once again regarding where the output is prettified

Reply:
In general, if the user specifies the flag `--format json`, the output will be "pretty" in a structured JSON format

Reply:
@FeynmanZhou ie there a reference to what prettified json that we can include or an expanded example with the outputs?

Reply:
@sajayantony The official [JSON spec](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiOrLXC-sCDAxWuhq8BHajEAwkQFnoECAYQAQ&url=https%3A%2F%2Fwww.json.org%2F&usg=AOvVaw3WUMhwoap01T91PbRZTt_w&opi=89978449) doesn't have definition on what prettified JSON means. 
I added a paragraph in FAQ to elaborate on the difference of prettified JSON and raw JSON in the ORAS context, see [line #324](https://github.com/oras-project/oras/pull/1199/files#diff-ddd85908fb8fee8c57a3726cb042b0572e9b4b333000356816268399ced78cccR324).